In [27]:
# Importing NumPy Library
import numpy as np

In [25]:
# Storing the data provided into a NumPy array which will facilitate statistical operations.
solar_radiation = np.array([11.1, 10.6, 6.3, 8.8, 10.7, 11.2, 8.9, 12.2])

In [26]:
# Computing the Mean, Median, and Variance in the respective order.

# Mean
print(f'{solar_radiation.mean()} is the current solar radiation mean.')

# Median
print(f'{np.median(solar_radiation):.3f} is the current solar radiation median.') # Limiting to 3 decimal places to standardize the measurements

# Variance
print(f'{np.var(solar_radiation):.3f} is the current solar radiation variance.') # Limiting to 3 decimal places to standardize the measurements


9.975 is the current solar radiation mean.
10.650 is the current solar radiation median.
3.084 is the current solar radiation variation.


In [30]:
# Second dataset to analyze statistical changes

# Adding 10 to each element of the dataset
sr10 = solar_radiation + 10

# Recalculating the same statistics (Mean, Median, and Variance)

# Mean
print(f'{sr10.mean()} is the current solar radiation after adding 10 to each observation mean.')

# Median
print(f'{np.median(sr10):.3f} is the current solar radiation after adding 10 to each observation median.') # Limiting to 3 decimal places to standardize the measurements

# Variance
print(f'{np.var(sr10):.3f} is the current solar radiation after adding 10 to each observation variance.') # Limiting to 3 decimal places to standardize the measurements

19.975 is the current solar radiation after adding 10 to each observation mean.
20.650 is the current solar radiation after adding 10 to each observation median.
3.084 is the current solar radiation after adding 10 to each observation variance.


## Short reflection

All the statistics changed as expected. Adding 10 to each observation makes the mean and median also increase by 10; however, the variance stays the same because the distance between each observation and the mean remains unchanged.

In [33]:
# Third dataset to analyze statistical changes

# Multiplying by -2 each element of the dataset
srm2 = solar_radiation * -2

# Recalculating the same statistics once again (Mean, Median, and Variance)

# Mean
print(f'{srm2.mean()} is the current solar radiation mean after multiplying each observation by negative 2.')

# Median
print(f'{np.median(srm2):.3f} is the current solar radiation median after multiplying each observation by negative 2.') # Limiting to 3 decimal places to standardize the measurements

# Variance
print(f'{np.var(srm2):.3f} is the current solar radiation variance after multiplying each observation by negative 2.') # Limiting to 3 decimal places to standardize the measurements

-19.95 is the current solar radiation mean after multiplying each observation by negative 2.
-21.300 is the current solar radiation median after multiplying each observation by negative 2.
12.337 is the current solar radiation variance after multiplying each observation by negative 2.


## Second short reflection

Now we can perceive a more abrupt change. The mean, which before was 9.975, is now -19.95, meaning that multiplying every observation by -2 also multiplies the mean by -2. The median, which before was 10.65, is now -21.30, following the same transformation. However, the variance behaves differently. The original variance was 3.084, and after multiplying every observation by -2, it increased to 12.337. This occurs because variance uses squared differences from the mean, so multiplying each observation by -2 causes the variance to be multiplied by (-2)², or 4.